include/odp/api/spec/traffic_mngr.h

```diff
@@ -1841,6 +1885,48 @@ typedef struct {
 int odp_tm_node_fanin_info(odp_tm_node_t             tm_node,
 			   odp_tm_node_fanin_info_t *info);
 
+/** The odp_tm_node_stats_t record type  is used to return various stats
+ * supported by a given tm_node via the odp_tm_node_stats() function.
+ */
+typedef struct odp_tm_node_stats_t {
+	/** Packets dropped by this node post scheduling/shaping at this node */
+	uint64_t dropped_pkts;
+
+	/** Octets dropped after scheduling/shaping at this node */
+	uint64_t dropped_octets;
+
+	/** Green packets that are sent through this tm node */
+	uint64_t green_pkts;
```

Comment:
I wonder if we need to or want to add these color specific counters now instead of single packet and byte counter for non-dropped packets of any color.

I am asking this because packet color related aspects of the TM API seem a bit incomplete to me and the API does not seem to allow the color (odp_packet_color(pkt)) of a packet to change when inside the TM system.

Reply:
Discussed with marvell transport team. Will drop the color specific stats from this PR and just add packets/octets counters.